Exercises/common.cmake

```diff
@@ -6,4 +6,29 @@ if(SPACK_CXX)
   set(ENV{CXX} ${SPACK_CXX})
 endif()
 
-find_package(Kokkos REQUIRED)
+set(Kokkos_COMMON_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../dep/Kokkos)
+
+include(FetchContent)
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.24)
+  FetchContent_Declare(
+    Kokkos
+    GIT_REPOSITORY https://github.com/kokkos/kokkos.git
+    GIT_TAG        4.0.01
+    SOURCE_DIR ${Kokkos_COMMON_SOURCE_DIR}
```

Comment:
My intent was to share the Kokkos source amongst all exercise but I suspect it does not work and it still attempt to download each time a new exercise is first configured